cmd/lakectl/cmd/local_commit.go

```diff
@@ -159,6 +166,11 @@ var localCommitCmd = &cobra.Command{
 			Repository: remote.Repository,
 			Ref:        remote.Ref,
 		}
+		DieOnErrorOrUnexpectedStatusCode(response, err, http.StatusCreated)
+		_, err = local.RemoveOperationFromIndexFile(localPath)
```

Comment:
This should be done immediately after Sync

Reply:
We're trying to protect the sync operation - not the commit. If we fail after the sync is completed there's no chance of data corruption